src/BenchmarksApps/HttpClientBenchmarks/Clients/HttpClient/Program.cs

```diff
@@ -0,0 +1,254 @@
+            BenchmarksEventSource.Log.Metadata("http/exceptions", "sum", "sum", "Exceptions", "Number of exceptions", "n0");
+            BenchmarksEventSource.Log.Metadata("http/rps/mean", "avg", "avg", "Mean RPS", "Requests per second - mean", "n0");
+
+            RegisterPercentiledMetric("http/headers", "Time to headers (ms)", "Time to headers (ms)");
+            RegisterPercentiledMetric("http/contentstart", "Time to first content byte (ms)", "Time to first content byte (ms)");
+            RegisterPercentiledMetric("http/contentend", "Time to last content byte (ms)", "Time to last content byte (ms)");
+
+            _headersTimes = new ConcurrentBag<long>();
+            _contentStartTimes = new ConcurrentBag<long>();
+            _contentEndTimes = new ConcurrentBag<long>();
+        }
+
+        using var cts = new CancellationTokenSource(TimeSpan.FromSeconds(duration));
+        var stopwatch = Stopwatch.StartNew();
+
+        var tasks = new List<Task>(_options.NumberOfClients * _options.ConcurrencyPerClient);
+        for (int i = 0; i < _options.NumberOfClients; ++i)
+        {
+            var client = _httpClients[i];
+            for (int j = 0; j < _options.ConcurrencyPerClient; ++j)
+            {
+                switch(_options.Scenario)
+                {
+                    case "get":
+                        tasks.Add(Get(client, cts.Token, collectMetrics));
+                        break;
+                    default:
+                        throw new ArgumentException($"Unknown scenario: {_options.Scenario}");
+                }
+            }
+        }
+
+        await Task.WhenAll(tasks);
+        var elapsed = stopwatch.Elapsed.TotalSeconds;
+
+        if (collectMetrics)
+        {
+            BenchmarksEventSource.Measure("http/successrequests", _successRequests);
+            BenchmarksEventSource.Measure("http/badstatusrequests", _badStatusRequests);
+            BenchmarksEventSource.Measure("http/exceptions", _exceptionRequests);
+            BenchmarksEventSource.Measure("http/rps/mean", (_successRequests + _badStatusRequests) / elapsed);
+
+            if (_successRequests > 0)
+            {
+                LogPercentiledMetric("http/headers", _headersTimes!, TicksToMs);
+                LogPercentiledMetric("http/contentstart", _contentStartTimes!, TicksToMs);
+                LogPercentiledMetric("http/contentend", _contentEndTimes!, TicksToMs);
+            }
+        }
+    }
+
+    private static async Task Get(HttpClient client, CancellationToken token, bool collectMetrics)
+    {
+        var oneByteArray = new byte[1];
+        var drainArray = new byte[81920];
+        var stopwatch = Stopwatch.StartNew();
+        while (!token.IsCancellationRequested)
+        {
+            try
+            {
+                stopwatch.Restart();
+                using var result = await client.GetAsync("/get", HttpCompletionOption.ResponseHeadersRead, token);
```

Comment:
Passing the cancellation token here means we will need to register for it on every request, which adds some amount of cost, possibly nontrivial.

While it's interesting to measure the cost of passing a cancellation token here, I'm not sure we want to do that in the base scenario.

It seems like it's better to just not pass the cancellation token and instead just let the call complete normally. We will see that the cancellation token is cancelled when the loop iterates.

Reply:
I passed CT to avoid hanging forever in case something will go wrong, but that actually will not be the case inside SendAsync because HttpClient timeout would still apply. I believe I would still need to use it later when reading response content. What do you think?

Reply:
Even if we set the timeout to infinite, HttpClient will still pass the `_pendingRequestsCts` here, which means we will still incur cancellation-related overhead.

We could choose to ignore cancellation here entirely. 

If we're not using any HttpClient features, could we use `HttpMessageInvoker(SocketsHttpHandler)` instead?
We'd still have the option to test what the overhead of cancellation is if we wanted to.

Reply:
The behavior of the underlying `CancellationTokenSource` could also be different here. With `HttpClient`, a single CTS will be contended by all requests. In cases like YARP, this would never be the case as each request has dedicated CTSs.

Reply:
I think it depends on what our goals are for the benchmark.  If it's to highlight best-case scenario that a highly specialized consumer like YARP will use, we could use HttpMessageInvoker.  If it's to highlight expected case for the vast majority of consumers, we should stick with HttpClient.  CancelPendingRequests is a good example a non-pay-for-play feature we should avoid adding in the future.

Reply:
We should start with scenarios that represent the "expected case for the vast majority of customers". This is the most valuable data since it best represents the performance of HttpClient in practice.

We can and should measure other, less common/more advanced scenarios or variations as well, later.